### PR TITLE
 Rework plugin search logic

### PR DESCRIPTION
**Pull request by mwestphal**
- Move plugin search paths logic to application
- Improve testing accordingly
- python binds have been checked and to not need improvements
- Update doc
- Rework CI logic for F3D_PLUGINS_PATH

### CODE REVIEW

File: doc/user/USAGE.md [hunk 1]
```diff
@@ -59,7 +59,7 @@ Here is how the plugins are searched (by precedence order):
 1. Search the static plugins
 2. Consider the option given is a full path
 3. Search in the paths specified in `F3D_PLUGINS_PATH` environment variable
-4. Search in the same directory than `f3d.exe` (Windows only)
+4. Search in a directory relative to the F3D application ( `./` on Windows, `../lib` otherwise).
```
Comment (Meakk) [inline doc/user/USAGE.md:62]: Is there any reason we are putting plugin DLLs in "bin" instead of "lib" on Windows?

File: library/public/engine.h [hunk 2]
```diff
@@ -112,15 +112,17 @@ class F3D_EXPORT engine
   /**
    * Load a plugin.
    * Supports full path, relative path, and plugin name.
-   * On Linux and macOS, uses LD_LIBRARY_PATH to find the plugin.
-   * On Windows, the plugin should be located in the same folder as the executable.
+   * First try to load the path provided as if it is a full path to a plugin.
```
Comment (Meakk) [inline library/public/engine.h:115]: Should we mention we are first trying to load a static plugin by name?

File: application/F3DOptionsParser.cxx [hunk 3]
```diff
@@ -188,6 +189,30 @@ class ConfigurationOptions
   std::string ExecutableName;
 };
 
+//----------------------------------------------------------------------------
+std::vector<std::string> ConfigurationOptions::GetPluginSearchPaths() const
+{
+#if F3D_MACOS_BUNDLE
+  return {};
+#endif
+#if !F3D_MACOS_BUNDLE
```
Comment (Meakk) [inline application/F3DOptionsParser.cxx:198]:
```suggestion
#else
```

**Review by Meakk — CHANGES_REQUESTED** [review]

File: library/public/engine.h [hunk 4]
```diff
@@ -112,15 +112,17 @@ class F3D_EXPORT engine
...
+   * Then try to load a plugin by its name looking into the provided plugin search paths.
+   * Then try to load a plugin by its name relying on internal system (eg: LD_LIBRARY_PATH).
    * The plugin "native" is always available and includes native VTK readers.
    * If built and available in your build, F3D is providing 5 additional plugins:
    * "alembic", "assimp", "draco", "exodus", "occt".
    * Custom plugins can also be available that F3D is not supporting officially.
    * Throw a plugin_exception if the plugin can't be loaded for some reason.
    */
-  static void loadPlugin(const std::string& path);
+  static void loadPlugin(
```
[the remark this answers is not in the document]
Comment (mwestphal) [reply library/public/engine.h:124]: nothing to update as stated in the description

File: application/testing/CMakeLists.txt [hunk 5]
```diff
@@ -177,12 +177,13 @@ f3d_test(NAME TestGroupGeometriesColoring DATA mb/recursive ARGS --group-geometr
 
 # Test missing symbol plugin using the libf3d
 if(WIN32)
-      f3d_test(NAME TestPluginInvalid ARGS --verbose --load-plugins "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/f3d.dll" NO_BASELINE REGEXP "Cannot find init_plugin symbol in library")
+  set(_libname "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/f3d.dll")
```
Comment (Meakk) [inline application/testing/CMakeLists.txt:180]: What about

```cmake
set(_libname ${CMAKE_SHARED_MODULE_PREFIX}f3d${CMAKE_SHARED_MODULE_SUFFIX})
```

File: application/testing/CMakeLists.txt [hunk 6]
```diff
@@ -397,11 +398,11 @@ if(F3D_PLUGIN_BUILD_EXODUS)
 
     # Test --load-plugins with a full path plugin
     if(WIN32)
-      set(_libname "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/f3d-plugin-exodus.dll")
+      set(_libname "f3d-plugin-exodus.dll")
```
Comment (Meakk) [inline application/testing/CMakeLists.txt:401]: here too

**Review by Meakk — CHANGES_REQUESTED** [review]

File: application/testing/CMakeLists.txt (hunk 5 above)
Comment (mwestphal) [reply application/testing/CMakeLists.txt:181]: what is the point ? this is not going to change and I cant factorise because of the `lib`

Comment (Meakk) [reply application/testing/CMakeLists.txt:181]: Please note the `${CMAKE_SHARED_MODULE_PREFIX}` being `lib` on Unix and empty on windows.

Comment (mwestphal) [reply application/testing/CMakeLists.txt:181]: !!

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -87,11 +87,8 @@ jobs:
       run: |
          mkdir build
          mkdir install
-
-    - name: Set F3D_PLUGINS_PATH to build
-      working-directory: ${{github.workspace}}
-      shell: bash
-      run: echo "F3D_PLUGINS_PATH=$(pwd)/build/lib" >> $GITHUB_ENV
+         cd install
+         mkdir output
```
Comment (Meakk) [inline .github/workflows/ci.yml:91]: mkdir install/output

**Pull request merged**